File: robert-server-common/src/main/java/fr/gouv/stopc/robertserver/common/RobertRequestType.java
package fr.gouv.stopc.robertserver.common;

import java.util.HashMap;
import java.util.Map;

public enum RobertRequestType {

    HELLO((byte) 0x01),
    STATUS((byte) 0x02),
    UNREGISTER((byte) 0x03),
    DELETE_HISTORY((byte) 0x04);

    private final byte salt;

    private static Map map = new HashMap<>();

    RobertRequestType(byte salt) {
        this.salt = salt;
    }

    public byte getValue() {
        return salt;
    }

    static {
        for (RobertRequestType digestSalt : RobertRequestType.values()) {
            map.put(digestSalt.salt, digestSalt);
        }
    }

    public static RobertRequestType valueOf(byte val) {
        return (RobertRequestType) map.get(val);
    }
}

File: robert-server-common/src/main/kotlin/fr/gouv/stopc/robertserver/common/RobertRequestType.kt
package fr.gouv.stopc.robertserver.common

/**
 * Robert protocol request type value.
 *
 * @see <a href="https://github.com/ROBERT-proximity-tracing/documents/blob/master/ROBERT-specification-EN-v1_1.pdf">Robert protocol</a> C. Authenticated Requests
 */
enum class RobertRequestType(val salt: Byte) {

    HELLO(1),
    STATUS(2),
    UNREGISTER(3),
    DELETE_HISTORY(4);

    companion object {

        @JvmStatic
        fun fromValue(salt: Int) = values().find { it.salt.toInt() == salt }

        @JvmStatic
        fun fromValue(salt: Byte) = values().find { it.salt == salt }
    }
}
